src/mmg3d/chkmsh_3d.c

```diff
@@ -305,6 +305,31 @@ void MMG3D_chkedgetag(MMG5_pMesh mesh, MMG5_int ip1, MMG5_int ip2, int tag) {
   }
 }
 
+/**
+ * \param mesh pointer to the mesh
+ *
+ * Check that faces do not have nonsensical tags (MG_GEO, MG_NOM, MG_CRN).
+ *
+ */
+void MMG3D_chkfacetags(MMG5_pMesh mesh) {
+  MMG5_pTetra  pt;
+  MMG5_pxTetra pxt;
+  MMG5_int     k;
+  int          i, tag;
+
+  for (k=1; k<=mesh->ne; k++) {
+    pt = &mesh->tetra[k];
+    if ( !MG_EOK(pt) )  continue;
+    if ( !pt->xt ) continue;
+
+    pxt = &mesh->xtetra[pt->xt];
+    for (i=0; i<4; i++) {
+      tag = pxt->ftag[i];
+      assert(!(tag & (MG_GEO & MG_NOM & MG_CRN)) && "Nonsensical tag on face");
```

Comment:
I think that here `(MG_GEO & MG_NOM & MG_CRN)` always evaluates to `0`. You should use the `|` bitwise operator instead of `&`: 
```
assert(!(tag & (MG_GEO | MG_NOM | MG_CRN))
```